### PR TITLE
Implemented scoring logic, also option to use weighted scoring

### PR DESCRIPTION
**Pull request by joshuasicw0818**

### CODE REVIEW

File: src/pose_estimation/scoring/angle_score.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    def compute_score(first_keypoints: KeypointStatistics, second_keypoints: KeypointStatistics, weights: np.ndarray = None) -> float:
+        """
+        Returns the average angle difference between
+        the two keypoints
+        :param first_keypoints: first set of keypoints
+        :param second_keypoints: second set of keypoints
+        :param weights: weights to apply to each angle
+        :return: angle score
+        """
+        angle_diff = np.abs(first_keypoints.to_list() - second_keypoints.to_list())
```
Comment (AlexWebb03) [inline src/pose_estimation/scoring/angle_score.py:17]: does `to_list` return a numpy array? Maybe you could add a `to_numpy` method in the `Keypoints` class for this to work?

File: src/pose_estimation/scoring/euclidean_score.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        the landmarks of the two keypoints
+        :param first_keypoints: first set of keypoints
+        :param second_keypoints: second set of keypoints
+        :param weights: weights to apply to each distance
+        :return: euclidean score
+        """
+        first_landmarks = np.array(first_keypoints.keypoints.normalized_landmarks)
+        second_landmarks = np.array(second_keypoints.keypoints.normalized_landmarks)
+
+        dist_difference = np.abs(first_landmarks - second_landmarks)
```
Comment (AlexWebb03) [inline src/pose_estimation/scoring/euclidean_score.py:21]: Think this should be
```
np.lingalg.norm(first_landmarks - second_landmarks)
```
to get the euclidean distance

**Review by AlexWebb03 — COMMENTED** [review]
Like the idea of weights!

File: src/pose_estimation/scoring/angle_score.py (hunk 1 above)
Comment (joshuasicw0818) [reply src/pose_estimation/scoring/angle_score.py:17]: Yea it returns a numpy array.

**Comment by AlexWebb03** [comment]
Thinking it would be good to test this, might add a main method if you don't mind @joshuasicw0818 ?

**Comment by joshuasicw0818** [comment]
> Thinking it would be good to test this, might add a main method if you don't mind @joshuasicw0818 ?

sure

**Review by AlexWebb03 — APPROVED** [review]

**Pull request merged**